boa/explorer.py

```diff
@@ -28,7 +28,7 @@ def _fetch_etherscan(
         res = SESSION.get(uri, params=params)
         res.raise_for_status()
         data = res.json()
-        if data.get("result") != "Max rate limit reached":
+        if data.get("status") == "0" and "rate limit" in data.get("result", ""):
```

Comment:
i think the conditional might be flipped? it is introducing some new test failures.

also, probably out of scope of this PR but maybe we should increase the backoff every time, to something like `1.1**i`

Reply:
```suggestion
        if not (data.get("status") == "0" and "rate limit" in data.get("result", "")):
```

Reply:
> also, probably out of scope of this PR but maybe we should increase the backoff every time, to something like `1.1**i`

https://github.com/vyperlang/titanoboa/pull/288